iwyu_preprocessor.cc

```diff
@@ -751,6 +758,23 @@ void IwyuPreprocessorInfo::FileChanged_ExitToFile(SourceLocation include_loc,
          "begin_exports without an end_exports");
     begin_exports_location_stack_.pop();
   }
+
+  const FileEntry* includer = GetFileEntry(associated_pragma_location_);
```

Comment:
Why does this new handling happen here in `FileChanged_ExitToFile` and not in `FileChanged_EnterFile` where the other associated header handling is? Is it because you want to reset the `associated_pragma_location` before leaving the includer file?

Also, I wanted to ask, because I'm not 100% clear on the code flow -- is this the only path where `associated_pragma_location` needs to be reset? We always end up getting a `FileChanged_ExitToFile` call if `HandlePragmaComment` has been called, right?